recipes/nvcloth/1.1.6/conanfile.py

```diff
@@ -0,0 +1,194 @@
+
+        cmake.definitions["TARGET_BUILD_PLATFORM"] = self._get_target_build_platform()
+
+        cmake.configure(
+            build_folder=os.path.join(self.build_folder, self._build_subfolder),
+            source_folder=os.path.join(self.build_folder),
+        )
+        return cmake
+    
+    def _remove_samples(self):
+        shutil.rmtree(os.path.join(self._source_subfolder, "NvCloth", "samples"))
+
+    def _copy_sources(self):
+        # Copy CMakeLists wrapper
+        shutil.copy(os.path.join(self.source_folder, "CMakeLists.txt"), "CMakeLists.txt")
+
+        # Copy patches
+        if "patches" in self.conan_data and not os.path.exists("patches"):
+            os.mkdir("patches")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            shutil.copy(os.path.join(self.source_folder, patch["patch_file"]), "patches")
+        
+        # Copy PhysX source code
+        subfolders_to_copy = [
+            "NvCloth",
+            "PxShared",
+        ]
+        for subfolder in subfolders_to_copy:
+            shutil.copytree(os.path.join(self.source_folder, self._source_subfolder, subfolder),
+                            os.path.join(self._source_subfolder, subfolder))
+    
+    def _patch_sources(self):
+        # There is no reason to force consumer of PhysX public headers to use one of
+        # NDEBUG or _DEBUG, since none of them relies on NDEBUG or _DEBUG
+        tools.replace_in_file(os.path.join(self.build_folder, self._source_subfolder, "PxShared", "include", "foundation", "PxPreprocessor.h"),
+                              "#error Exactly one of NDEBUG and _DEBUG needs to be defined!",
+                              "// #error Exactly one of NDEBUG and _DEBUG needs to be defined!")
+        shutil.copy(
+            os.path.join(self.build_folder, self._source_subfolder, "NvCloth/include/NvCloth/Callbacks.h"),
+            os.path.join(self.build_folder, self._source_subfolder, "NvCloth/include/NvCloth/Callbacks.h.origin")
+        )
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        
+        if self.settings.build_type == "Debug":
+            shutil.copy(
+                os.path.join(self.build_folder, self._source_subfolder, "NvCloth/include/NvCloth/Callbacks.h"),
+                os.path.join(self.build_folder, self._source_subfolder, "NvCloth/include/NvCloth/Callbacks.h.patched")
+            )
+            shutil.copy(
+                os.path.join(self.build_folder, self._source_subfolder, "NvCloth/include/NvCloth/Callbacks.h.origin"),
+                os.path.join(self.build_folder, self._source_subfolder, "NvCloth/include/NvCloth/Callbacks.h")
+            )
+
+    def build(self):
+        os.environ['GW_DEPS_ROOT'] = os.path.abspath(self._source_subfolder)
+        self._copy_sources()
+        self._patch_sources()
+        self._remove_samples()
+        cmake = self._configure_cmake()
+        cmake.build()
```

Comment:
```suggestion
        with tools.environment_append({"GW_DEPS_ROOT": os.path.abspath(self._source_subfolder)}):
            self._patch_sources()
            self._remove_samples()
            cmake = self._configure_cmake()
            cmake.build()
```
Use something for environment variable: https://docs.conan.io/en/latest/reference/tools.html#tools-environment-append